File: postm/routes/posts.py
from flask.blueprints import Blueprint

from postm.controllers.posts import PostsController
from postm.middlewares.authentication import AuthMiddleware

postsBp = Blueprint(
    name = "posts",
    import_name = __name__,
    url_prefix = "/posts" 
)

postsController = PostsController()

middleware = AuthMiddleware()

@postsBp.before_request
def postsMiddleware():
    return middleware.auth()

@postsBp.post("/create")
def createPost():
    return postsController.create()

@postsBp.get("/all")
def findAllPost():
    return postsController.findAll()

@postsBp.get("/find/<string:id>")
def findPostById(id: str):
    return postsController.findById(id)

@postsBp.delete("/delete/<string:id>")
def deletePost(id: str):
    return postsController.delete(id)

@postsBp.put("/update/<string:id>")
def updatePost(id: str):
    return postsController.update(id)

@postsBp.get("/all/page")
def findAllPostPaged():
    return postsController.findAllPaged()

File: src/routes/posts.ts
import { Router, Request, Response } from "express";
import Controller from "../controllers/posts";

import upload from "../middlewares/upload";
import Authentication from "../middlewares/authentication";

const controller = new Controller();
const router = Router();
const userAuth = new Authentication();

router.post(
    "/create", 
    upload.single("image"), 
    controller.create
);

router.get("/all", userAuth.auth, controller.all);

router.get("/find/:id", userAuth.auth, controller.find);

router.delete("/delete/:id", userAuth.auth, controller.delete);

router.put(
    "/update/:id", 
    userAuth.auth,
    upload.single("image"), 
    controller.update
);

router.get("/all/page", userAuth.auth, controller.allPaged);

export default router;
